sources/packages/backend/apps/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -626,21 +610,55 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
       .where("studentScholasticStanding.reversalDate IS NULL")
       .andWhere("student.id = :studentId", {
         studentId,
-      })
-      .groupBy("application.offeringIntensity")
-      .getRawMany<ScholasticStandingSummary>();
+      });
 
-    const scholasticStandingSummaryPartTime = scholasticStandingSummary.find(
-      (summary) => summary.offeringIntensity === OfferingIntensity.partTime,
-    );
-    const scholasticStandingSummaryFullTime = scholasticStandingSummary.find(
-      (summary) => summary.offeringIntensity === OfferingIntensity.fullTime,
-    );
-    return {
-      partTimeUnsuccessfulCompletionWeeks:
-        scholasticStandingSummaryPartTime?.totalUnsuccessfulWeeks ?? 0,
-      fullTimeUnsuccessfulCompletionWeeks:
-        scholasticStandingSummaryFullTime?.totalUnsuccessfulWeeks ?? 0,
-    };
+    if (offeringIntensity) {
+      queryBuilder.andWhere(
+        "application.offeringIntensity = :offeringIntensity",
+        {
+          offeringIntensity,
+        },
+      );
+      // If offeringIntensity is specified, return only the sum for that intensity.
+      const result = await queryBuilder
+        .groupBy("application.offeringIntensity")
+        .getRawOne<ScholasticStandingSummary>();
+      // If fullTime, add SFAS weeks.
+      if (offeringIntensity === OfferingIntensity.fullTime) {
+        const sfasUnsuccessfulCompletionWeeks =
+          await this.sfasIndividualService.getSFASTotalUnsuccessfulCompletionWeeks(
+            studentId,
+          );
+        return {
+          fullTimeUnsuccessfulCompletionWeeks:
+            sfasUnsuccessfulCompletionWeeks +
+            (result?.totalUnsuccessfulWeeks ?? 0),
+        };
+      }
+    } else {
```

Comment:
Looks like there is a misunderstanding from dev syncup regarding the filter.

I will not consider using the filter as a blocker. But if the filter is used, it must be used only to restrict the FT legacy query which is always required as of now. 

Hence, it would recommend not to change any logic or introduce filter but just centralize the logic.